### PR TITLE
Add reference test

### PR DESCRIPTION
**Pull request by oliver----**
- render command names as classes for tools and menu items as a preparation for UI tests
- test opening and closing the 'Add Reference' workflow
- test creating each ref type

### CODE REVIEW

**Comment by oliver----** [comment]
@Integral ready for you to take over. In some cases we might want to add CSS selectors in the components so that it is easier or cleaner to access buttons etc.

**Comment by Integral** [comment]
@oliver---- could you please help me with configuring async tests map?
[Here](https://github.com/substance/texture/pull/681/commits/45ced556751bc190ba860b258b7d563a7a32b6f2#diff-a0d7fc3a13b214fa7c0fa690995b8eb3R24) I'm trying to run a small test suite against each type of reference, I'm pretty sure we need this because in the past we have some of them broken.
So I'm trying to map those tests for each ref type and run. This is even working, but in a very strange way: it's working slowly and I have a felling that it resolves after some timeout, not with the test ending... Could you look into it?

File: src/article/shared/AddReferenceWorkflow.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ export default class AddReferenceWorkflow extends Component {
     const refTypesButtons = $$('ul').addClass('se-reftypes-list')
     targetTypes.forEach(item => {
       refTypesButtons.append(
-        $$('li').addClass('se-type').append(
+        $$('li').addClass('se-type sm-' + item).append(
```
Comment (oliver----) [inline src/article/shared/AddReferenceWorkflow.js]: Right. These things we need for UI tests.

**Comment by oliver----** [comment]
@Integral This was looking pretty good already 👍 
I think the only 'missing' piece was an 'await' for the call of the actually async test impl.

**Comment by oliver----** [comment]
Feel free to merge!

**Comment by oliver----** [comment]
There seems to be an issue with MemoryDOM. Need to fix that before we can merge.

**Comment by oliver----** [comment]
@Integral merging this

**Pull request merged**